test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -561,17 +566,43 @@ scopes[upgrade_to_latest]="upgrade/downgrade databases to the latest versions of
 scopes[wt_standalone]="WiredTiger standalone releases"
 scopes[two_versions]="any two given versions"
 
-# The following associative array maps the 'official' build system to use for each branch.
-# CMake build support is reliably mature in newer release branches, whilst earlier revisions
-# primarily use Autoconf (note: some earlier branches may have CMake support, but these aren't
-# considered 'mature' versions.)
-declare -A build_sys
-build_sys['develop']="cmake"
-build_sys['mongodb-6.0']="cmake"
-build_sys['mongodb-5.0']="autoconf"
-build_sys['mongodb-4.4']="autoconf"
-build_sys['mongodb-4.2']="autoconf"
-build_sys['mongodb-4.0']="autoconf"
+#############################################################
+# Retrive the build system used by a particular release,
+# since WiredTiger switched from autoconf to cmake.
+#       arg1: branch name to check against
+#       output: string as name of build system
+#############################################################
+get_build_system()
+{
+    branch="$1"
+    # Default to cmake.
+    local build_system="cmake"
+
+    # As of MongoDB 6.0, WiredTiger standalone builds switched to CMake
+    if [[ $branch == mongodb-* ]]; then
+        major=`echo $branch | cut -d '-' -f 2 | cut -d '.' -f 1`
+        if [ $major -lt 6 ]; then
+            build_system="autoconf"
+        fi
+    fi
+    # Check for WiredTiger standalone branch names. They are of the form 10.0.0 - figure out
+    # if the first element is numerical, and use that to decide.
+    wt_version="false"
+    major=`echo $branch | cut -d '.' -f 1`
+    case $major in
+        ''|*[!0-9]*) wt_version="false" ;;
+        *) wt_version="true" ;;
+    esac
+
+    if [[ "$wt_version" == "true" ]]; then
```

Comment:
A slightly cleaner way to do this would be a regex check
```bash
if [[ "$wt_version" =~ "[0-9]+" ]]; then
```
and we can drop `wt_version` and
```bash
    case $major in
        ''|*[!0-9]*) wt_version="false" ;;
        *) wt_version="true" ;;
    esac
```
above

Reply:
I thought about using a regular expression, but wasn't sure if it was available on all the platforms this needs to run, so stuck with something clunkier. If we use regex in the shell scripts other places I agree it's cleaner

Reply:
Yeah good point. I think this is a bash-ism and not available on all shells.
We use it in a few places in the code base, including line 490 of this file, but I'm happy to leave the code as is

Reply:
Thanks. It's likely just something I'm out of date on, but it seems sort of urgent to get this in (despite me letting it languish for 9 days)